x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/results/utils.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Result } from '../../../result/types';
+import { CurationResult } from '../../types';
+
+/**
+ * The `promoted` and `hidden` keys from the internal curations endpoints
+ * currently return a document data structure that our Result component can't
+ * correctly parse - we need to attempt to naively transform the data in order
+ * to display it in a Result
+ *
+ * TODO: Ideally someday we can update our internal curations endpoint to return
+ * the same Result-ready data structure that the `organic` endpoint uses, and
+ * remove this file when that happens
+ */
```

Comment:
Just want to highlight this whole file 😬 

This was the original issue I discussed in our sync a week or two ago where the documents coming back for the `promoted` and `hidden` lists were not in the standard format that our Result component wants (the `organic` list was in the correct format however).

This is in a separate `utils.ts` file from the top-level one because I'm hoping we can fix this at the server-level someday and then just delete this entire file / helpers entirely.